### PR TITLE
WT-6499 Ignore evict priority for btrees that are dominating cache usage

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
As we create more collections, the metadata file grows. In a system with lots (multiple thousands) of dhandles but barely any updates, the metadata file can dominate cache usage.

In the case where the metadata file causes the cache usage to go past any of the targets, the eviction server will kick in. But it will intentionally ignore the metadata file due to its evict priority until it exceeds the trigger and we begin evicting aggressively. In an idle system where cache usage isn't going up and we never begin evicting aggressively, the eviction server will continuously look for pages to evict without any success leading to high CPU usage.

My change is to ignore the evict priority if the file is currently contributing to half or more of the target cache usage. This seems like a sensible rule generally speaking and it does solve the high CPU usage we were seeing in this workload.

### CODE REVIEW

File: src/include/cache.i [hunk 1]
```diff
@@ -352,6 +352,32 @@ __wt_eviction_updates_needed(WT_SESSION_IMPL *session, double *pct_fullp)
...
+    bytes_max = S2C(session)->cache_size + 1;
+
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_inmem) >
+      (uint64_t)(0.5 * cache->eviction_target * bytes_max) / 100)
+        return (true);
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_dirty_intl + btree->bytes_dirty_leaf) >
+      (uint64_t)(0.5 * cache->eviction_dirty_target * bytes_max) / 100)
+        return (true);
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_updates) >
+      (uint64_t)(0.5 * cache->eviction_updates_target * bytes_max) / 100)
```
Comment (tetsuo-cpp) [inline src/include/cache.i:375]: In this particular instance, it was the `eviction_updates_target` that caused the eviction server to kick in. But I think this rule should apply to any of the targets.

**Comment by sulabhM** [comment]
I had a chat with Alex C., though the change as presented here looks good, an area of exploration is to reduce complexity by getting rid of the eviction priority altogether.

**Comment by tetsuo-cpp** [comment]
> I had a chat with Alex C., though the change as presented here looks good, an area of exploration is to reduce complexity by getting rid of the eviction priority altogether.

Agreed.
I still think that this should go in before we do that exploratory work to remove eviction priority.
I wouldn't expect a perf degradation from a change like this but I wouldn't be surprised if we saw issues with removing eviction priority since I assume we added it for a reason.

**Comment by sulabhM** [comment]
> I still think that this should go in before we do that exploratory work to remove eviction priority.
> I wouldn't expect a perf degradation from a change like this but I wouldn't be surprised if we saw issues with removing eviction priority entirely since I'm assuming we added that to address a type of workload.

Agreed. Your change as it is today should only affect metadata, and I think it is good to start evicting metadata if it consumes half of the targets that we have set. I briefly looked at the code, metadata might be the only file getting affected by eviction priority. If we remove the consideration for priority and not add another mechanism to keep metadata in memory, we will end up eviting metadata and then having to read it back again as needed for access of other tables.

I think we should check in this code and create a separate ticket for simplifying if needed.

**Comment by tetsuo-cpp** [comment]
> > I still think that this should go in before we do that exploratory work to remove eviction priority.
> > I wouldn't expect a perf degradation from a change like this but I wouldn't be surprised if we saw issues with removing eviction priority entirely since I'm assuming we added that to address a type of workload.
> 
> Agreed. Your change as it is today should only affect metadata, and I think it is good to start evicting metadata if it consumes half of the targets that we have set. I briefly looked at the code, metadata might be the only file getting affected by eviction priority. If we remove the consideration for priority and not add another mechanism to keep metadata in memory, we will end up eviting metadata and then having to read it back again as needed for access of other tables.
> 
> I think we should check in this code and create a separate ticket for simplifying if needed.

Sounds good. I created WT-6541 to track this work.

**Comment by tetsuo-cpp** [comment]
@agorrod Friendly ping. Can we get this in soon?

**Comment by tetsuo-cpp** [comment]
@agorrod 
Sys-perf testing looked ok for the most part but got disrupted by the ongoing Evergreen outages.
I'll restart it later tonight and take a look at the results + Jenkins perf testing next week.

**Pull request merged**